### PR TITLE
Lpinheiro/chore/migrate openai models

### PR DESCRIPTION
**Pull request by lspinheiro**
## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: python/packages/autogen-core/src/autogen_core/components/models/__init__.py [hunk 1]
```diff
@@ -1,23 +1,19 @@
...
     FunctionExecutionResultMessage,
     LLMMessage,
     RequestUsage,
     SystemMessage,
+    TopLogprob,
     UserMessage,
 )
 
 __all__ = [
-    "AzureOpenAIChatCompletionClient",
```
Comment (ekzhu) [inline python/packages/autogen-core/src/autogen_core/components/models/__init__.py:19]: We will also need to show deprecation warnings when user attempt to import from `autogen_core.components.models`

Comment (jackgerrits) [reply python/packages/autogen-core/src/autogen_core/components/models/__init__.py:19]: Massive +1 to this

Comment (lspinheiro) [reply python/packages/autogen-core/src/autogen_core/components/models/__init__.py:19]: Done. Couldn't remove the code and make a conditional import due to circular dependencies but the warning was added

**Comment by lspinheiro** [comment]
Replaced by #3848

**Pull request closed without merging**